drivers/SPI.cpp

```diff
@@ -36,23 +36,23 @@ SPI::SPI(PinName mosi, PinName miso, PinName sclk, PinName ssel) :
     // No lock needed in the constructor
 
     spi_init(&_spi, mosi, miso, sclk, ssel);
-    aquire();
+    _acquire();
 }
 
 void SPI::format(int bits, int mode) {
     lock();
     _bits = bits;
     _mode = mode;
-    SPI::_owner = NULL; // Not that elegant, but works. rmeyer
-    aquire();
+    spi_format(&_spi, _bits, _mode, 0);
+    _owner = this;
     unlock();
 }
 
 void SPI::frequency(int hz) {
     lock();
     _hz = hz;
-    SPI::_owner = NULL; // Not that elegant, but works. rmeyer
-    aquire();
+    spi_frequency(&_spi, _hz);
+    _owner = this;
```

Comment:
The owner can be changed here without ensuring that both frequency and format are set. The format setting maybe be incorrect on the next write since it was not updated.

A sequence similar to this would cause that:
```
spi_a.write(value_a)                 <-Internal call to _acquire will change owner and set frequency and format for SPI A
spi_b.frequency(freq_b)              <- Owner changed to B and frequency set
spi_b.write(value_b)                 <- Interal call to _acquire won't change setting. Frequency will be set correctly for B and format will be set for A
```